feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,769 @@
+	for _, item := range list {
+		if item == target {
+			return true
+		}
+	}
+	return false
+}
+
+func removeElement(list []string, element string) []string {
+	for i := 0; i < len(list); i++ {
+		if list[i] == element {
+			list = append(list[:i], list[i+1:]...)
+			i-- // Adjust index to account for removed element
+		}
+	}
+	return list
+}
+
+func TestCheckForCoreFiles(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	timestamp := uint64(time.Now().UTC().Unix())
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	coreFileCheck(t, dut, gnoiClient, timestamp, true)
+}
+
+func TestComponentStatus(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	fabricCards := components.FindComponentsByType(t, dut, fabricCardType)
+	checkComponents := append(controllerCards, lineCards...)
+	checkComponents = append(checkComponents, fabricCards...)
+	// check oper-status of the components is Active.
+	for _, component := range checkComponents {
+		val, present := gnmi.Lookup(t, dut, gnmi.OC().Component(component).OperStatus().State()).Val()
+		if !present {
+			t.Errorf("ERROR: Get component %s oper-status failed", component)
+		} else {
+			t.Logf("INFO: Component %s oper-status: %s", component, val)
+		}
+	}
+
+	if deviations.ConsistentComponentNamesUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation consistent_component_names_unsupported")
+	}
+
+	// use gNOI to check components health status is not "unhealthy".
+	gnoiClient := dut.RawAPIs().GNOI().New(t)
+	for _, component := range checkComponents {
+		componentName := map[string]string{"name": component}
+		req := &hpb.GetRequest{
+			Path: &tpb.Path{
+				Elem: []*tpb.PathElem{
+					{Name: "components"},
+					{
+						Name: "component",
+						Key:  componentName,
+					},
+				},
+			},
+		}
+		validResponse, err := gnoiClient.Healthz().Get(context.Background(), req)
+		if err != nil {
+			t.Errorf("Error: %v", err)
+			continue
+		}
+		t.Logf("INFO: Component %s Healthz Status: %s", component, validResponse.GetComponent().Status)
+	}
+}
+
+func TestControllerCardsNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	if deviations.ControllerCardCPUUtilizationUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation controller_card_cpu_utilization_unsupported")
+	}
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	cpuCards := components.FindComponentsByType(t, dut, cpuType)
+	for _, cpu := range cpuCards {
+		query := gnmi.OC().Component(cpu).State()
+		timestamp := time.Now().Round(time.Second)
+		component := gnmi.Get(t, dut, query)
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		if contains(controllerCards, cpuParent) {
```

Comment:
can we use slices.Contains here? 

https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
Thanks for your comments. Update code as below:

```go
	if slices.Contains(controllerCards, cpuParent) {
```
```go
	if slices.Contains(lineCards, cpuParent) {
```